domain/usecases/list_topics_test.go

```diff
@@ -0,0 +1,103 @@
+package usecases
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/matheusmosca/walrus/domain/entities"
+	"github.com/matheusmosca/walrus/domain/vos"
+)
+
+func TestListTopics(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		ctx context.Context
+	}
+
+	type fields struct {
+		storage Repository
+	}
+
+	tests := []struct {
+		name      string
+		args      args
+		fields    fields
+		beforeRun func(topic entities.Topic) chan vos.Message
+		want      []vos.TopicName
+		wantErr   bool
```

Comment:
could you change this to be an error type? and check this error with ErrorIs? The second test which the repository returns `errors.New("wrong")` could be deleted.

Reply:
done